Automattic/pocket-casts-android
# Migrate MediaBrowserService to MediaLibraryService

Parent #887

## Description

This PR 
- Registers both (`MediaBrowserService`, `MediaLibraryService`) service interfaces for backward compatibility - 2040f7b7afed93a6ea4726245fe032418624ae50
- Adds `media3-session` dependency - 5389e811bab6f1c61e4eef89f27a6725c4ea28c3
- Uses `androidx.media3.common.MediaItem` instead of [MediaBrowserCompat.MediaItem](https://developer.android.com/reference/kotlin/android/support/v4/media/MediaBrowserCompat.MediaItem) 
d82b1ff3554e9ca772eaec9202d5ac98b6267a0a
- Implements `MediaLibraryService.MediaLibrarySessionCallback` - [diff](https://github.com/Automattic/pocket-casts-android/commit/d82b1ff3554e9ca772eaec9202d5ac98b6267a0a#diff-9cfa2a8a05b3d2dbd76dd8db7d068a1ea5103a000fbb6b194c2079c544736adbR604-R672)
    - Uses `Futures` in contrast to the detachable `Result` approach of the `MediaBrowserServiceCompat` (Adds [kotlin-coroutines-guava](https://github.com/Automattic/pocket-casts-android/commit/d82b1ff3554e9ca772eaec9202d5ac98b6267a0a#diff-d979b641bd0ea7c9da3fe113f9657636df1002652c75042cfe3b5203da064215R213) dependency to makes the `MediaBrowser` API fit more naturally into a suspend method as recommended [here](https://developer.android.com/guide/topics/media/media3/getting-started/migration-guide#MediaBrowser).)
-  Changes service to inherit from a `MediaLibraryService` instead of `MediaBrowserService` - 
- Implement the service life-cycle methods
   - Overrides `onGetSession(MediaSession.ControllerInfo)` to return the `MediaLibrarySession` that was built in `onCreate`.
- Implements `MediaSession.Callback.onAddMediaItems()`
- Migrates `AutoPlaybackService`

## Testing instructions
- Make sure that intermediate code-changes are as described above
- App compiles and episode can be played

Note that: 
- Lint check is disabled temporarily while changes are in progress 
- Changes to initialize the player correctly, control media playback in the session and more will be added as part of upcoming PRs.
